hatch_pip_compile/plugin.py

```diff
@@ -91,10 +91,10 @@ def get_option_types() -> Dict[str, Any]:
         return {  # pragma: no cover
             "lock-filename": str,
             "pip-compile-hashes": bool,
-            "pip-compile-args": List[str],
+            "pip-compile-args": list,
```

Comment:
Could we have this as `list[str]` with a `from __future__ import annotations`?

Reply:
Thanks for trying! I see this is how the VirtualEnvironment is implementing this as well.

https://github.com/pypa/hatch/blob/4ebce0e1fe8bf0fcdef587a704c207a063d72575/src/hatch/env/virtual.py#L130-L132